src/SymfonyCache/TaggableStore.php

```diff
@@ -0,0 +1,387 @@
+        ];
+
+        // If the response has a Vary header we remove the non-varying entry
+        if ($response->hasVary()) {
+            unset($entries[self::NON_VARYING_KEY]);
+        }
+
+        // Support tagging
+        $tags = [];
+        if ($response->headers->has($this->purgeTagsHeader)) {
+            $tags = explode(',', $response->headers->get($this->purgeTagsHeader));
+        }
+
+        $this->save($cacheKey, $entries, $tags);
+
+        return $cacheKey;
+    }
+
+    /**
+     * Invalidates all cache entries that match the request.
+     *
+     * @param Request $request A Request instance
+     */
+    public function invalidate(Request $request)
+    {
+        $cacheKey = $this->getCacheKey($request);
+
+        $this->cache->deleteItem($cacheKey);
+    }
+
+    /**
+     * Locks the cache for a given Request.
+     *
+     * @param Request $request A Request instance
+     *
+     * @return bool|string true if the lock is acquired, the path to the current lock otherwise
+     */
+    public function lock(Request $request)
+    {
+        $cacheKey = $this->getCacheKey($request);
+
+        if (isset($this->locks[$cacheKey])) {
+            return false;
+        }
+
+        $this->locks[$cacheKey] = $this->lockFactory
+            ->createLock($cacheKey);
+
+        return $this->locks[$cacheKey]->acquire();
+    }
+
+    /**
+     * Releases the lock for the given Request.
+     *
+     * @param Request $request A Request instance
+     *
+     * @return bool False if the lock file does not exist or cannot be unlocked, true otherwise
+     */
+    public function unlock(Request $request)
+    {
+        $cacheKey = $this->getCacheKey($request);
+
+        if (!isset($this->locks[$cacheKey])) {
+            return false;
+        }
+
+        try {
+            $this->locks[$cacheKey]->release();
+        } catch (LockReleasingException $e) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns whether or not a lock exists.
+     *
+     * @param Request $request A Request instance
+     *
+     * @return bool true if lock exists, false otherwise
+     */
+    public function isLocked(Request $request)
+    {
+        $cacheKey = $this->getCacheKey($request);
+
+        if (!isset($this->locks[$cacheKey])) {
+            return false;
+        }
+
+        return $this->locks[$cacheKey]->isAcquired();
+    }
+
+    /**
+     * Purges data for the given URL.
+     *
+     * @param string $url A URL
+     *
+     * @return bool true if the URL exists and has been purged, false otherwise
+     */
+    public function purge($url)
+    {
+        $cacheKey = $this->getCacheKey(Request::create($url));
+
+        return $this->cache->deleteItem($cacheKey);
+    }
+
+    /**
+     * Cleanups storage.
+     */
+    public function cleanup()
+    {
+        /** @var LockInterface $lock */
+        foreach ($this->locks as $lock) {
+            $lock->release();
```

Comment:
you should wrap in a try/catch block

Reply:
True, will add!

Reply:
Added in 2f4b53497692b953fced5505ae5a151789419962.